plugins/optimization-detective/detect.js

```diff
@@ -154,35 +154,52 @@ function getGroupForViewportWidth( viewportWidth, urlMetricGroupStatuses ) {
  * @param {string}               currentETag          - Current ETag.
  * @param {string}               currentUrl           - Current URL.
  * @param {URLMetricGroupStatus} urlMetricGroupStatus - URL Metric group status.
- * @return {Promise<string>} Session storage key.
+ * @param {Logger}               logger               - Logger.
+ * @return {Promise<string|null>} Session storage key for the current URL or null if crypto is not available or caused an error.
  */
 async function getAlreadySubmittedSessionStorageKey(
 	currentETag,
 	currentUrl,
-	urlMetricGroupStatus
+	urlMetricGroupStatus,
+	{ warn, error }
 ) {
-	const message = [
-		currentETag,
-		currentUrl,
-		urlMetricGroupStatus.minimumViewportWidth,
-		urlMetricGroupStatus.maximumViewportWidth || '',
-	].join( '-' );
+	if ( ! window.crypto || ! window.crypto.subtle ) {
```

Comment:
Nit: only checking `window.crypto` should suffice. There's no combination where the former is available but the latter isn't.

Reply:
Nevertheless someone could try to polyfill `crypto.getRandomValues()` but not define `crypto.subtle`.

Reply:
> So like so?

Yes that's what I meant.

> Nevertheless someone could try to polyfill `crypto.getRandomValues()` but not define `crypto.subtle`.

Sounds unlikely to me but if we are concerned about that then we should keep the check.